src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+				onChange();
+				return onError?.(errs);
+			}
+
+			acceptedFiles.forEach((file) => {
+				const reader = new FileReader();
+				reader.onabort = () => console.log('file reading was aborted');
+				reader.onerror = () => {
+					const err = [`Failed to upload ${file.name}`];
+					onChange();
+					setErrors(err);
+					onError?.(err);
+				};
+				reader.onloadstart = () => setIsLoading(true);
+				reader.onprogress = (event) => {
+					setLoadingPercentage(Math.floor((100 * event.loaded) / event.total));
+				};
+				reader.onload = () => {
+					if (file != null && reader.result != null) {
+						setIsLoading(false);
+						onChange({
+							dataUrl: reader.result as string,
+							uploadedFile: file,
+						});
+					}
+				};
+
+				reader.readAsDataURL(file);
+			});
+		}) as NonNullable<DropzoneOptions['onDrop']>,
+		[],
+	);
+
+	const { getRootProps, getInputProps, isDragActive } = useDropzone({
+		onDrop,
+		maxSize,
+		accept,
+		multiple: false,
+		noDrag: false,
+		disabled: isLoading,
+	});
+
+	return (
+		<>
+			{props.schema.title && <InputLabel>{props.schema.title}</InputLabel>}
+			<Box
+				sx={{
+					border: (theme) =>
+						`dashed 2px ${
+							errors.length !== 0
+								? '#F19B9E'
+								: isDragActive
+								? theme.palette.primary.main
+								: 'rgba(0, 0, 0, 0.23)'
+						}`,
+					borderRadius: '2px',
+					display: 'flex',
+					flexDirection: 'column',
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+				style={{ minHeight: '150px' }}
+				flexDirection="column"
+				alignItems="center"
+				justifyContent="center"
+				{...getRootProps()}
+				dir={undefined}
+			>
+				<input {...getInputProps()} />
+
+				{isLoading && (
+					<Box
+						sx={{
+							position: 'absolute',
+							top: 0,
+							left: 0,
+							width: '100%',
+							height: '100%',
+							display: 'flex',
+							alignItems: 'center',
+							justifyContent: 'center',
+							background: 'rgba(255, 255, 255, 0.7)',
+						}}
+					>
+						<CircularProgress variant="determinate" value={loadingPercentage} />
+					</Box>
+				)}
+
+				<Box display="flex" alignItems="center" justifyContent="center">
+					<Box>
+						{value ? (
+							isImage(value) ? (
+								<Avatar
+									sx={{
+										borderRadius: 0,
+										height: '100px',
+										width: '100px',
+										backgroundSize: 'contain',
+									}}
+									src={dataUrl(value)}
+								/>
+							) : (
+								<FilePresentIcon
+									sx={{
+										fontSize: 64,
+										color: (theme) => theme.palette.grey[500],
+									}}
+								/>
+							)
+						) : (
+							<UploadFileIcon
+								sx={{ fontSize: 64, color: (theme) => theme.palette.grey[500] }}
+							/>
+						)}
+					</Box>
+					<Box>
+						<Typography>
+							<Typography variant="inherit">Drag and drop or</Typography>{' '}
+							<Typography variant="inherit">
+								<Link>browse</Link> files
+							</Typography>
+						</Typography>
```

Comment:
Is there a reason this wouldn't work? Also cc @drskullster should this text be pretending to be a `Link`? Or should we have a `plain` Button style?
```suggestion
						<Typography>Drag and drop or <Link>browse</Link> files</Typography>
```

Reply:
Also wondering why there is no `onClick` for the `browse`? Isn't clicking it supposed to open the user's documents for browsing and uploading?

Reply:
Now that is indeed an impotant question @myarmolinsky because right now the whole area is clickable (if you click anywhere it will be prompted to browse a file). I have no idea which behaviour we want in here really